PurchasesTests/Purchasing/OfferingsManagerTests.swift

```diff
@@ -100,24 +100,114 @@ extension OfferingsManagerTests {
         expect(obtainedOfferings).to(beNil())
     }
 
-    func testOfferingsForAppUserIDReturnsUnexpectedBackendErrorIfBadBackendRequest() {
+    func testOfferingsForAppUserIDReturnsConfigurationErrorIfBackendReturnsEmpty() throws {
+        // given
+        mockBackend.stubbedGetOfferingsCompletionResult = ([:], nil)
+        mockOfferingsFactory.emptyOfferings = true
+
+        // when
+        var obtainedOfferings: Offerings?
+        var completionCalled = false
+        var obtainedError: Error?
+        offeringsManager.offerings(appUserID: MockData.anyAppUserID) { maybeOfferings, maybeError in
+            obtainedOfferings = maybeOfferings
+            completionCalled = true
+            obtainedError = maybeError
+        }
+
+        // then
+        expect(completionCalled).toEventually(beTrue())
+        expect(obtainedOfferings).to(beNil())
+        let error = try XCTUnwrap(obtainedError)
+        expect((error as NSError).code) == ErrorCode.configurationError.rawValue
+    }
+
+    func testOfferingsForAppUserIDReturnsConfigurationErrorIfProductsRequestsReturnsEmpty() throws {
```

Comment:
i.e.: configuration error if app store connect or storekit config file have no products for the offerings in the backend